opentracing/tracecontext.go

```diff
@@ -43,11 +48,18 @@ type TraceContext interface {
 	// wish to use them as HTTP header keys (or key suffixes), and of course
 	// HTTP headers are case insensitive.
 	//
+	// `restrictedKey` MUST match the regular expression
+	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. See
+	// CanonicalizeTraceTagKey().
+	//
```

Comment:
The dapperish implementation panics if the key is invalid, should all implementations do that? If so, I think it'd be good ot document it here too.